cfscrape/__init__.py

```diff
@@ -61,6 +64,19 @@ def is_cloudflare_challenge(self, resp):
         )
 
     def request(self, method, url, *args, **kwargs):
+        self.headers = (
```

Comment:
Right here you are overwriting the existing headers that are present. For example, after sticking a debugger in here above this line. `self.headers` already exists and has contents.

```ipdb> c
> /app/sickrage/lib/cfscrape/__init__.py(70)request()
     69         self.headers = (
---> 70             OrderedDict(
     71                 [

ipdb> self.headers
{'Connection': 'keep-alive', u'Content-Type': u'application/json', u'Accept-Encoding': u'gzip,deflate', 'Accept': '*/*', u'User-Agent': u'SickChill.CE.1/(Linux; 4.18.20-unRAID; 469f26e1-6018-11e9-9ad7-0242ac110002)'}
```

For my request to be successful, I need the `Content-Type` header to remain. Other than that - the commit works well,  hopefully we can get this in soon.

Reply:
On second thought, could you explain this change? It also changes some other headers that are needed, I also assume that, for the servers that respect it, we don't want to limit them to English. The 'Accept' also may varry.

Reply:
@CodyWoolaver These headers seemed to be working best with Cloudflare, but I agree that breaking support for custom headers doesn't make sense. Can you share a site where you have problems with my current version.

The following gist contains a version which tries to restore support for custom headers by only using best working headers as long as cloudflare has not been bypassed:
https://gist.github.com/lukele/ce188004545192c0d92064e85138f0ab

Reply:
Perhaps you could set those headers for specifically interacting with cloudflare and interactions with the intended site. That or you could merge in the specific headers you have picked and that way the intended headers could still make it through.

The site that this caused issues for was actually a local install of deluge and their webapi. There is a hard requirement to have Content-Type when interacting with it (no idea why). The client accessing it is https://github.com/SickChill/SickChill/tree/master/lib/cfscrape and uses a copy of cfscrape.

Let me know if I can help in any way.

Reply:
> Perhaps you could set those headers for specifically interacting with cloudflare and preserve the interactions with the intended site. 

That's basically what I'm doing in the gist. The only problem is that Cloudflare did require for example the user-agent not to change between requests, so merging might be more problematic. Did you try the gist by any chance?

Reply:
That makes sense - to preserve the User-Agent as it is a variable that should not frequently change. Is there any risk with picking a couple of headers and making them stay that way?

For example force the header values of `User-Agent Connection Upgrade-Insecure-Requests` to be consistent and do `kwargs['headers'].update(<contents>)` and that way the user agent, and other important headers remain and we can still pass in custom headers.

I have not had a chance to test it out yet, I won't be in a position to do so for several hours. I can get back to you then.

Reply:
I would also like to mention, that I completely removed lines 67-79 of this PR in the code running at home and was able to get through without a problem.

Reply:
Any more movement on this @lukele? The proposed fix you provided did work, however I still believe we should not be completely ignoring the headers if they are passed in. Having a set of headers that we require to go to cloudflare makes sense, but it should be an update on the object, not a replacement of.
@Anorov do you have any input on this?

Reply:
@CodyWoolaver sorry for the late reply. Best have a look at cloudscraper (https://github.com/VeNoMouS/cloudscraper). It‘s based on cfscrape but more actively maintained for now. It also includes some good fixes regarding the sometimes presented captcha challenge.